File: Data Structure/String/Easy/LET_no67_Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int n1 = a.size(), n2 = b.size();
        int n = max(n1, n2) + 1;
        string res(n, '0');
        int i = n1 - 1, j = n2 - 1, k = n - 1, cur = 0;
        while (i >= 0 || j >= 0) {
            if (i >= 0)
                cur += a[i--] - '0';
            if (j >= 0)
                cur += b[j--] - '0';
            res[k--] = '0' + cur % 2;
            cur /= 2;
        }
        res[k] = '0' + cur % 2;
        return res[0] == '0' ? res.substr(1) : res;
    }
};

File: Data Structure/String/Easy/LET_no67_Add Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        m, n = len(a), len(b)
        res = []
        cur = 0
        i, j = m - 1, n - 1
        while i >= 0 or j >= 0:
            if i >= 0:
                cur += int(a[i])
                i -= 1
            if j >= 0:
                cur += int(b[j])
                j -= 1
            res.append(str(cur % 2))
            cur //= 2
        
        if cur > 0:
            res.append(str(cur))
        
        return ''.join(res[::-1])